waddrmgr/scoped_manager.go

```diff
@@ -2324,6 +2321,20 @@ func (s *ScopedKeyManager) importScriptAddress(ns walletdb.ReadWriteBucket,
 	// Add the new managed address to the cache of recent addresses and
 	// return it.
 	s.addrs[addrKey(scriptIdent)] = managedAddr
+
+	if updateStartBlock {
+		// Now that the database has been updated, update the start block in
+		// memory too if needed. Ensure the manager lock goes outside the scoped
+		// manager lock.
+		s.mtx.Unlock()
```

Comment:
Since we're releasing the lock here, it's possible for a second/parallel call to `importScriptAddress` to come in and potentially progress to the `if bs.Height < s.rootManager.syncState.startBlock.Height` check before the below is updated, similar to how we could end up with duplicated addresses in https://github.com/lightningnetwork/lnd/issues/8697. BUT, since we're just increasing the start block, the worst that could happen is that this write would happen twice. So IMO an acceptable solution until we do a full rewrite of this logic (which I agree is definitely necessary at some point).